scripts/setup_osx.sh

```diff
@@ -1,38 +1,267 @@
-#!/bin/bash
-# Make sure that pip is available
-hash brew 2>/dev/null || {
-  echo "Please install homebrew before continuing. You can use the" \
-    "following command to install:"
-  echo "/usr/bin/ruby -e \"\$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)\""
-  exit 0
+#!/usr/bin/env bash
+# This script installs garage on OS X distributions.
+#
+# NOTICE: To keep consistency across this script, scripts/setup_linux.sh and
+# docker/Dockerfile.ci, if there's any changes applied to this file, specially
+# regarding the installation of dependencies, apply those same changes to the
+# mentioned files.
+
+# Exit if any error occurs
+set -e
+
+# Add OS X versions where garage is successfully installed in this list
+VERIFIED_OSX_VERSIONS=("10.12")
+
+### START OF CODE GENERATED BY Argbash v2.6.1 one line above ###
+die()
+{
+  local _ret=$2
+  test -n "$_ret" || _ret=1
+  test "$_PRINT_HELP" = yes && print_help >&2
+  echo "$1" >&2
+  exit ${_ret}
 }
 
-hash conda 2>/dev/null || {
-  echo "Please install anaconda before continuing. You can download" \
-    "it at https://www.continuum.io/downloads. Please use the Python 2.7" \
-    " installer."
-  exit 0
+begins_with_short_option()
+{
+  local first_option all_short_options
+  all_short_options='h'
+  first_option="${1:0:1}"
+  test "$all_short_options" = "${all_short_options/$first_option/}" && \
+    return 1 || return 0
 }
 
 
-echo "Installing system dependencies"
-echo "You will probably be asked for your sudo password."
 
-brew install swig sdl sdl_image sdl_mixer sdl_ttf portmidi openmpi
+# THE DEFAULTS INITIALIZATION - POSITIONALS
+_positionals=()
+# THE DEFAULTS INITIALIZATION - OPTIONALS
+_arg_mjkey=
+_arg_modify_bashprof="off"
 
-# Make sure that we're under the directory of the project
-cd "$(dirname "$0")/.."
+print_help ()
+{
+  printf '%s\n' "Installer of garage for OS X."
+  printf 'Usage: %s [--mjkey <arg>] [--(no-)modify-bashprof] ' "$0"
+  printf '[-h|--help]\n'
+  printf '\t%s\n' "--mjkey: Path of the MuJoCo key (no default)"
+  printf '\t%s' "--modify-bashprof,--no-modify-bashprof: Set environment "
```

Comment:
Please keep the arg naming consistent across scripts. --modify-bashrc is fine on mac, even though it's actually in bash_profile.